includes/admin/shortcodes/class-shortcode-button.php

```diff
@@ -32,26 +32,34 @@
 	 * Class constructor
 	 */
 	public function __construct() {
-		add_action( 'admin_init', array( $this, 'init'), 999 );
+		add_action( 'current_screen', array( $this, 'init' ), 999 );
+		add_action( 'admin_init', array( $this, 'ajax_handler' ) );
 	}
 
 	/**
 	 * Initialize
 	 *
-	 * @since 2.1.0
+	 * @since  2.1.0
 	 * @access public
 	 */
-	public function init(){
+	public function init() {
 		if ( $this->is_add_button() ) {
 			add_filter( 'mce_external_plugins', array( $this, 'mce_external_plugins' ), 15 );
 
 			add_action( 'admin_enqueue_scripts', array( $this, 'admin_enqueue_assets' ) );
 			add_action( 'admin_enqueue_scripts', array( $this, 'admin_localize_scripts' ), 13 );
 			add_action( 'media_buttons', array( $this, 'shortcode_button' ) );
 		}
+	}
 
+
+	/**
+	 * Ajax handler for shortcode
+	 *
+	 * @since 2.3.0
+	 */
+	public function ajax_handler(){
 		add_action( "wp_ajax_give_shortcode", array( $this, 'shortcode_ajax' ) );
-		add_action( "wp_ajax_nopriv_give_shortcode", array( $this, 'shortcode_ajax' ) );
```

Comment:
@DevinWalker Remove non logged in ajax handler because we are rendering shortcode button only in backend